gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -48,166 +49,185 @@
 
 public final class BaselineErrorProne implements Plugin<Project> {
     private static final Logger log = Logging.getLogger(BaselineErrorProne.class);
-
-    public static final String REFASTER_CONFIGURATION = "refaster";
     public static final String EXTENSION_NAME = "baselineErrorProne";
-
     private static final String ERROR_PRONE_JAVAC_VERSION = "9+181-r4173-1";
     private static final String PROP_ERROR_PRONE_APPLY = "errorProneApply";
     private static final String PROP_REFASTER_APPLY = "refasterApply";
 
-    @SuppressWarnings("UnstableApiUsage")
+    private Project project;
+    private BaselineErrorProneExtension errorProneExtension;
+    private Task compileRefaster;
+    private Provider<File> refasterRulesFile;
+
     @Override
-    public void apply(Project project) {
-        project.getPluginManager().withPlugin("java", plugin -> {
-            BaselineErrorProneExtension errorProneExtension = project.getExtensions()
-                    .create(EXTENSION_NAME, BaselineErrorProneExtension.class, project);
-            project.getPluginManager().apply(ErrorPronePlugin.class);
-
-            String version = Optional.ofNullable(getClass().getPackage().getImplementationVersion())
-                    .orElse("latest.release");
-
-            Configuration refasterConfiguration = project.getConfigurations().create(REFASTER_CONFIGURATION);
-            Configuration refasterCompilerConfiguration = project.getConfigurations()
-                    .create("refasterCompiler", configuration -> configuration.extendsFrom(refasterConfiguration));
-
-            project.getDependencies().add(
-                    REFASTER_CONFIGURATION,
-                    "com.palantir.baseline:baseline-refaster-rules:" + version + ":sources");
-            project.getDependencies().add(
-                    ErrorPronePlugin.CONFIGURATION_NAME,
-                    "com.palantir.baseline:baseline-error-prone:" + version);
-            project.getDependencies().add(
-                    "refasterCompiler",
-                    "com.palantir.baseline:baseline-refaster-javac-plugin:" + version);
-
-            Provider<File> refasterRulesFile = project.getLayout().getBuildDirectory()
-                    .file("refaster/rules.refaster")
-                    .map(RegularFile::getAsFile);
-
-            Task compileRefaster = project.getTasks().create("compileRefaster", RefasterCompileTask.class, task -> {
-                task.setSource(refasterConfiguration);
-                task.getRefasterSources().set(refasterConfiguration);
-                task.setClasspath(refasterCompilerConfiguration);
-                task.getRefasterRulesFile().set(refasterRulesFile);
-            });
+    public void apply(Project proj) {
+        proj.getPluginManager().withPlugin("java", plugin -> {
+            applyToJavaProject(proj);
```

Comment:
I had to do this refactor because cyclomatic complexity dinged us. Happy to put it in a separate PR if you like, otherwise it's just the last commit to read.